app/src/main/kotlin/com/simplemobiletools/flashlight/screens/SettingsScreen.kt

```diff
@@ -0,0 +1,114 @@
+package com.simplemobiletools.flashlight.screens
+
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import com.simplemobiletools.commons.compose.extensions.MyDevices
+import com.simplemobiletools.commons.compose.settings.SettingsCheckBoxComponent
+import com.simplemobiletools.commons.compose.settings.SettingsGroup
+import com.simplemobiletools.commons.compose.settings.SettingsPreferenceComponent
+import com.simplemobiletools.commons.compose.settings.SettingsTitleTextComponent
+import com.simplemobiletools.commons.compose.settings.scaffold.SettingsScaffold
+import com.simplemobiletools.commons.compose.theme.AppThemeSurface
+import com.simplemobiletools.commons.compose.theme.divider_grey
+import com.simplemobiletools.commons.helpers.isTiramisuPlus
+import com.simplemobiletools.flashlight.R
+
+@Composable
+internal fun SettingsScreen(
+    displayLanguage: String,
+    turnFlashlightOnStartupChecked: Boolean,
+    forcePortraitModeChecked: Boolean,
+    showBrightDisplayButtonChecked: Boolean,
+    showSosButtonChecked: Boolean,
+    showStroboscopeButtonChecked: Boolean,
+    onSetupLanguagePress: () -> Unit,
+    customizeColors: () -> Unit,
+    customizeWidgetColors: () -> Unit,
+    onTurnFlashlightOnStartupPress: (Boolean) -> Unit,
+    onForcePortraitModePress: (Boolean) -> Unit,
+    onShowBrightDisplayButtonPress: (Boolean) -> Unit,
+    onShowSosButtonPress: (Boolean) -> Unit,
+    onShowStroboscopeButtonPress: (Boolean) -> Unit,
+    goBack: () -> Unit,
```

Comment:
Way too many parameters, try slicing them into `SettingsGroups` maybe?

Reply:
Yeah, forgot to slice this screen :smile: 
Makes sense, I think it should be enough.

Reply:
GeneralSection is still fairly big, but it didn't make much sense to split it further.